### PR TITLE
Test : Included tests for build.go

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Included tests for build.go

Signed-off-by: naveensrinivasan <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

**Comment by naveensrinivasan** [comment]
@laurentsimon  👀

File: internal/builders/go/pkg/build_test.go [hunk 1]
```diff
@@ -1047,3 +1048,114 @@ func Test_generateCommand(t *testing.T) {
...
+		dry bool
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "valid flags",
```
Comment (ianlewis) [inline internal/builders/go/pkg/build_test.go:1069]: Add whether the test is dry or not to the name

File: internal/builders/go/pkg/build_test.go [hunk 2]
```diff
@@ -1047,3 +1048,114 @@ func Test_generateCommand(t *testing.T) {
...
+					Goos:    "linux",
+					Goarch:  "amd64",
+					Binary:  "binary",
+					Main:    asPointer("../builders/go/main.go"),
+					Dir:     asPointer("../builders/go"),
+					Ldflags: []string{},
+				},
+			},
+			args: args{
+				dry: true,
```
Comment (ianlewis) [inline internal/builders/go/pkg/build_test.go:1133]: Do you want to maybe add non-dry versions of these tests? or do they need to be dry for other reasons?

File: internal/builders/go/pkg/build_test.go [hunk 3]
```diff
@@ -1047,3 +1048,114 @@ func Test_generateCommand(t *testing.T) {
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &GoBuild{
+				cfg:     tt.fields.cfg,
+				goc:     tt.fields.goc,
+				argEnv:  tt.fields.argEnv,
+				ldflags: tt.fields.ldflags,
+			}
+			defer os.Unsetenv("OUT_BINARY")
+			os.Setenv("OUTPUT_BINARY", tt.fields.cfg.Binary)
```
Comment (ianlewis) [inline internal/builders/go/pkg/build_test.go:1147]: Use t.Setenv so that the env var is cleaned up automatically.

**Review by ianlewis — APPROVED** [review]
Thanks!

File: internal/builders/go/pkg/build_test.go [hunk 4]
```diff
@@ -1047,3 +1048,114 @@ func Test_generateCommand(t *testing.T) {
...
+		ldflags string
+	}
+	type args struct {
+		dry bool
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
```
Comment (laurentsimon) [inline internal/builders/go/pkg/build_test.go:1066]: Could you use the expected error `error err` instead of `bool`?

**Review by asraa — APPROVED** [review]
Thanks! LGTM

Wondering as well (non-blocking), for dry=false, can we look for the output binary to make sure we build something?

File: internal/builders/go/pkg/build_test.go (hunk 4 above)
Comment (naveensrinivasan) [reply internal/builders/go/pkg/build_test.go:1066]: I also included the error

File: internal/builders/go/pkg/build_test.go (hunk 3 above)
Comment (naveensrinivasan) [reply internal/builders/go/pkg/build_test.go]: Agreed. Thanks

File: internal/builders/go/pkg/build_test.go (hunk 1 above)
Comment (naveensrinivasan) [reply internal/builders/go/pkg/build_test.go]: I took care of that. Thanks

**Comment by naveensrinivasan** [comment]
> Thanks! LGTM
> 
> Wondering as well (non-blocking), for dry=false, can we look for the output binary to make sure we build something?

As I mentioned in the previous comment I will do that in the follow up PR.

File: internal/builders/go/pkg/build_test.go (hunk 4 above)
Comment (laurentsimon) [reply internal/builders/go/pkg/build_test.go:1066]: you should be able to remove the `wantErr` altogether as well?

File: internal/builders/go/pkg/build_test.go [hunk 5]
```diff
@@ -1047,3 +1048,123 @@ func Test_generateCommand(t *testing.T) {
...
+			}
+			err := b.Run(tt.args.dry)
+			if err != nil != tt.wantErr {
+				t.Errorf("Run() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if tt.err != nil {
+				if err == nil {
+					t.Errorf("Run() error = nil, wantErr %v", tt.err)
+				} else if err.Error() != tt.err.Error() {
+					t.Errorf("Run() error = %v, wantErr %v", err, tt.err)
```
Comment (ianlewis) [inline internal/builders/go/pkg/build_test.go:1165]: Also to be consistent with other tests you can use `cmp.Diff(err, tt.err)` in the error message.

File: internal/builders/go/pkg/build_test.go [hunk 6]
```diff
@@ -1047,3 +1048,123 @@ func Test_generateCommand(t *testing.T) {
...
+				b.goc = path
+			}
+			err := b.Run(tt.args.dry)
+			if err != nil != tt.wantErr {
+				t.Errorf("Run() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if tt.err != nil {
+				if err == nil {
+					t.Errorf("Run() error = nil, wantErr %v", tt.err)
+				} else if err.Error() != tt.err.Error() {
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply internal/builders/go/pkg/build_test.go]: I took care of it.

File: internal/builders/go/pkg/build_test.go (hunk 5 above)
Comment (naveensrinivasan) [reply internal/builders/go/pkg/build_test.go]: I addressed it.

**Pull request merged**